apps/provisioning/pkg/jobs/validator.go

```diff
@@ -0,0 +1,190 @@
+package jobs
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	provisioning "github.com/grafana/grafana/apps/provisioning/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/apps/provisioning/pkg/repository/git"
+	"github.com/grafana/grafana/apps/provisioning/pkg/safepath"
+)
+
+// ValidateJob performs validation on the Job specification and returns an error if validation fails
+func ValidateJob(job *provisioning.Job) error {
+	list := field.ErrorList{}
+
+	// Validate action is specified
+	if job.Spec.Action == "" {
+		list = append(list, field.Required(field.NewPath("spec", "action"), "action must be specified"))
+		return toError(job.Name, list) // Early return since we can't validate further without knowing the action
+	}
+
+	// Validate action is a valid JobAction
+	validActions := []provisioning.JobAction{
+		provisioning.JobActionPull,
+		provisioning.JobActionPush,
+		provisioning.JobActionPullRequest,
+		provisioning.JobActionMigrate,
+		provisioning.JobActionDelete,
+		provisioning.JobActionMove,
+	}
+	actionValid := false
+	for _, va := range validActions {
+		if job.Spec.Action == va {
+			actionValid = true
+			break
+		}
+	}
+	if !actionValid {
+		list = append(list, field.Invalid(field.NewPath("spec", "action"), job.Spec.Action, "invalid action"))
+	}
+
+	// Validate repository is specified
+	if job.Spec.Repository == "" {
+		list = append(list, field.Required(field.NewPath("spec", "repository"), "repository must be specified"))
+	}
+
+	// Validate action-specific options
+	switch job.Spec.Action {
```

Comment:
Related to the `validActions` check above, I guess we could have the same identical resource if we have a `default` clause here and append an error on the list in case the `switch` falls there?